lib/membrane_mp4/payloader/h264.ex

```diff
@@ -34,11 +34,16 @@ defmodule Membrane.MP4.Payloader.H264 do
     {nalus, metadata} = process_metadata(metadata)
     nalus = Enum.map(nalus, &Map.put(&1, :payload, :binary.part(payload, &1.unprefixed_poslen)))
 
+    new_caps =
+      if Enum.any?(nalus, &(&1.metadata.h264.type == :pps)),
+        do: generate_caps(ctx.pads.input.caps, nalus),
+        else: ctx.pads.output.caps
+
     caps =
-      if ctx.pads.output.caps do
-        []
-      else
+      if ctx.pads.output.caps != new_caps do
         [caps: {:output, generate_caps(ctx.pads.input.caps, nalus)}]
+      else
+        []
       end
```

Comment:
let's cache and compare SPS and PPS themselves before we generate and compare the caps

Reply:
Do you mean the `Enum.any?` case? To check if the `PPS` or `SPS` changed and if so then generate the new caps?

Reply:
yes